cache/demux.go

```diff
@@ -137,27 +134,24 @@ func (d *demux) resyncLaggingWatchers() {
 	defer d.mu.Unlock()
 
 	for w, nextRev := range d.laggingWatchers {
-		var oldestRev int64
-		if oldestEvent := d.history.PeekOldest(); oldestEvent != nil {
-			oldestRev = oldestEvent.Kv.ModRevision
-		}
+		oldestRev := d.history.PeekOldest()
 
 		if oldestRev != 0 && nextRev < oldestRev {
 			w.Stop()
 			delete(d.laggingWatchers, w)
 			continue
 		}
 		// TODO: re-enable key‐predicate in Filter when non‐zero startRev or performance tuning is needed
-		missedEvents := d.history.Filter(AfterRev(nextRev))
+		missed := d.history.Filter(AfterRev(nextRev))
 
-		for _, event := range missedEvents {
-			if !w.enqueueEvent(event) { // buffer overflow: watcher still lagging
+		for _, eventBatch := range missed {
+			if !w.enqueueEvent(eventBatch) { // buffer overflow: watcher still lagging
 				break
 			}
-			nextRev = event.Kv.ModRevision + 1
+			nextRev = eventBatch[0].Kv.ModRevision + 1
 		}
 
-		if len(missedEvents) > 0 && nextRev > missedEvents[len(missedEvents)-1].Kv.ModRevision {
+		if len(missed) > 0 && nextRev > missed[len(missed)-1][0].Kv.ModRevision {
```

Comment:
Looks like I missed this. The `len(missed) > 0` check is incorrect, as watchers that are don't have any missing event will not be marked as synchronized.